Neuro/Pathfinding/DataStructures/Node.cs

```diff
@@ -38,4 +36,15 @@ public int CompareTo(Node other)
         if (compare == 0) compare = hCost.CompareTo(other.hCost);
         return -compare;
     }
+
+    public virtual Node Clone()
+    {
+        return new Node(WorldPosition, GridPosition, IsAccessible)
+        {
+            Color = Color,
+            TransportSelfId = TransportSelfId,
+            TransportTargetId = TransportTargetId,
+            TransportNeighborsCache = TransportNeighborsCache
+        };
```

Comment:
This will not create a fully deep clone because Lists are reference types. Here is the solution I used in another project:

```suggestion
        Node clonedNode = new(WorldPosition, GridPosition, IsAccessible)
        {
            Color = Color,
            TransportSelfId = TransportSelfId,
            TransportTargetId = TransportTargetId,
            TransportNeighborsCache = new List<Node>(TransportNeighborsCache.Capacity)
        };
        clonedNode.TransportNeighborsCache.AddRange(TransportNeighborsCache);
        
        return clonedNode;
```